azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/StorageAccountApiLiveTest.java

```diff
@@ -0,0 +1,134 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.azurecompute.arm.domain.CreateStorageServiceParams;
+import org.jclouds.azurecompute.arm.domain.StorageService;
+import org.jclouds.azurecompute.arm.domain.StorageServiceKeys;
+import org.jclouds.azurecompute.arm.domain.StorageServiceUpdateParams;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+@Test(groups = "live", testName = "StorageAccountApiLiveTest")
+public class StorageAccountApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private static final String NAME = String.format("%3.24s",
+           RAND + StorageAccountApiLiveTest.class.getSimpleName().toLowerCase());
+
+   private void check(final StorageService storage) {
+      assertNotNull(storage.id());
+      assertNotNull(storage.name());
+      assertNotNull(storage.storageServiceProperties());
+      assertNotNull(storage.storageServiceProperties().accountType());
+      assertFalse(storage.storageServiceProperties().primaryEndpoints().isEmpty());
+      assertNotNull(storage.storageServiceProperties().creationTime());
+   }
+
+   @Test()
+   public void testList() {
+      List<StorageService> storages = api().list();
+      assertTrue(storages.size() > 0);
+      for (StorageService storage : storages) {
+         check(storage);
+      }
+   }
+
+   @Test()
+   public void testListAll() {
+      List<StorageService> storages = api().list();
+      assertTrue(storages.size() > 0);
+      for (StorageService storage : storages) {
+         check(storage);
+      }
+   }
+
+   @Test()
+   public void testIsAvailable() {
+      assertTrue(api().isAvailable(NAME).nameAvailable().equals("true"));
+   }
+
+   @Test(dependsOnMethods = "testIsAvailable")
+   public void testCreate() {
+
+      CreateStorageServiceParams storage = api().create(NAME, LOCATION, ImmutableMap.of("property_name",
+              "property_value"), ImmutableMap.of("accountType", StorageService.AccountType.Standard_ZRS.toString()));
+      while (storage == null) {
+         try {
+            Thread.sleep(25 * 1000);
+         } catch (InterruptedException e) {
+            e.printStackTrace();
+         }
+         storage = api().create(NAME, LOCATION, ImmutableMap.of("property_name", "property_value"),
+                 ImmutableMap.of("accountType", StorageService.AccountType.Standard_ZRS.toString()));
+      }
```

Comment:
Which are the semantics of the create operations? It looks kinda weird to attempt the create operation N times instead of calling a `get` operation when waiting.